### PR TITLE
Move containerd fs package in tree

### PR DESCRIPTION
**Pull request by cpuguy83**
This allows the fs package to be versioned separately from containerd
and makes for a nice interface for filesystem interactions in
continuity.

### CODE REVIEW

**Comment by cpuguy83** [comment]
I have not yet setup the Makefile for root tests ala containerd.

**Comment by stevvooe** [comment]
LGTM on first pass.

**Comment by justincormack** [comment]
Ah good, I need to find some time to work on this now...

**Comment by cpuguy83** [comment]
For some reason we need sudo to run the fs tests.
copy_file_range is getting EPERM otherwise.... could be related to a seccomp policy or something in the travis container config?

**Comment by cpuguy83** [comment]
Ok, this is green.

**Comment by AkihiroSuda** [comment]
LGTM, I think we should reorganize README to cover non-proto packages like this

**Comment by justincormack** [comment]
Sudo doesnt affest seccomp policy so no...

**Comment by cpuguy83** [comment]
@justincormack Well, when `sudo` is used on travis you get a full VM rather than a container.

**Comment by justincormack** [comment]
Ah. Well almost certainly then they are running a seccomp policy/docker version that is older than their kernel, so doesnt know about `copy_file_range` and blocks it, while the kernel does support it..

**Comment by dmcgowan** [comment]
Can you drop the `github.com/containerd/containerd` files now that they are no longer needed. I see you already removed the errdefs requirement and removed from the vendor file.

**Comment by cpuguy83** [comment]
Good catch, done.

File: fs/copy.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+var bufferPool = &sync.Pool{
+	New: func() interface{} {
+		buffer := make([]byte, 32*1024)
+		return &buffer
+	},
+}
+
+// CopyDir copies the directory from src to dst.
+// Most efficient copy of files is attempted.
+func CopyDir(dst, src string) error {
```
[the remark this answers is not in the document]
Comment (dmcgowan) [reply fs/copy.go]: Let's save this discussion. We need to do a 3 way merge of this function, with one Tonis wrote, and one in Moby. Yes we should rename it, but we can do that after we move it.

[the remark this answers is not in the document]
Comment (dmcgowan) [reply fs/copy.go]: Let me elaborate a bit more...
- This function only handles doing a direct copy of directories, attempts to use copy file range to speed things up
- Moby function is also doing a directory copy but other optimizations have been added to make sure the copy is fast and there is a an open PR to speed it up more with parallelizing the slower parts.
- Tonis has a similar copy function he uses for build kit but can also apply filters (and maybe works on more than just directories?)

[the remark this answers is not in the document]
Comment (stevvooe) [reply fs/copy.go]: @dmcgowan File a bug for that. @ijc has opinions there, as well.

**Pull request merged**